cmd/minishift/cmd/stop.go

```diff
@@ -40,16 +41,29 @@ VM but does not delete any associated files. To start the cluster again, use the
 }
 
 func runStop(cmd *cobra.Command, args []string) {
-	fmt.Println("Stopping local OpenShift cluster...")
 	api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
 	defer api.Close()
 
+	// if VM does not exist, exit with error
+	util.ExitIfUndefined(api, constants.MachineName)
+
+	hostVm, err := api.Load(constants.MachineName)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+
+	// check if VM is already in stopped state
+	if util.IsHostStopped(hostVm.Driver) {
```

Comment:
> There is a semantic difference. ExitIfNotRunning will just check for a running VM. There are more states than "stopped" and "running". If we use ExitIfNotRunning, Minishift won't try to attempt to stop if the VM is for example in state Error (which is probably should).
> 
> I think it is right to not use ExitIfNotRunning(). Whether it is enough to use check against "stopped" state is a different question. I think it will do though.

Right, there seems to be a lot of state. I think it is fine with just check if it is already stopped.

```
var states = []string{
	"",
	"Running",
	"Paused",
	"Saved",
	"Stopped",
	"Stopping",
	"Starting",
	"Error",
	"Timeout",
}
```